### DISCUSSION TITLE
How to use this repository as the primary branch for local development

### DISCUSSION THREAD

**Discussion by joshuafontany** [TiddlyWiki/TiddlyWiki5#6385] [General]
I found this very useful, and had to hunt around for it again, so thought I would post it here as a discussion.

How to setup this repository as your `master` branch & work out of local branches (that are backed up to github):

https://github.com/Jermolene/TiddlyWiki5/pull/5899#issuecomment-886597320

@pmario wrote:

I don't use my local "master" branch anymore. I directly use jermolene/master as my master branch. With that setting I also avoid to merge stuff into my master, because I can't push it to github... It would want to push it to Jeremy's branch, which I can't... So I'm sure I always create a new feature branch for new stuff.

Since I use VSCode all the command line stuff is hidden by the UI.

My "master-branch" setting in .git/config looks like this
```
[remote "jermolene"]
	url = https://github.com/Jermolene/TiddlyWiki5.git
	fetch = +refs/heads/*:refs/remotes/jermolene/*

...

[branch "master"]
	remote = jermolene
	merge = refs/heads/master
```
This is the only [branch "master"] setting! So if I checkout "master" I do get the actual version from Jeremy.

To get the latest version I need:
```
git pull --tags jermolene master
```

**Comment by joshuafontany** [comment]
A simple example. First, fork this repository to https://github.com/<UserName>/TiddlyWiki5.git:

```
cd /
mkdir tw5
cd tw5
mkdir TiddlyWiki5
git clone https://github.com/Jermolene/TiddlyWiki5.git ./TiddlyWiki5
cd TiddlyWiki5
git remote rename origin jermolene
git remote add origin https://github.com/<UserName>/TiddlyWiki5.git
git checkout -b dev
```
Make changes in the dev branch and
```
git push
```
If changes are in the upstream remote ("jermolene"), pull them, push them to your master remote ("origin"), then merge them into the dev branch (& push that):
```
git checkout master
git pull --tags jermolene master
git push
git checkout dev
git merge master
git push
```

I hope this is helpful. :P

**Reply by saqimtiaz** [reply]
> If changes are in the upstream remote ("jermolene"), pull them, push them to your master remote ("origin")

Note that pushing to your master remote ("origin") is not necessary as long as you aren't creating branches from it from the Github UI itself.

**Reply by pmario** [reply]
>Note that pushing to your master remote ("origin") is not necessary as long as you aren't creating branches from it from the Github UI itself.

That's true. That's an advantage. There is no need to keep origin master up to date anymore. Merging master to origin creates the "nasty" commits that are visible in PRs. ... 

My workflow is as follows atm: 

 - Switch to (Jeremy's) master branch
 - pull master and tags if needed
 - checkout a new dev-branch
 - push new dev-branch to origin
 - Create PR with GitHub Web UI

For local dev I'm using VSCode now. So the whole thing is done with VSCode UI -- I only use the command line if something goes wrong. ...

**Comment by Jermolene** [comment]
Hi @joshuafontany this is useful – would you consider making a contribution to /dev? It needs some love, but there's no harm in adding to the "howto's" that we have already.
